### PR TITLE
PIM-9393: Avoid user to remove all permissions to edit a job instance

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

User should not be allowed to remove all the permissions to edit a job instance at least one should be selected.
Event is triggered in EE to check if permission is not empty

=> EE PR : https://github.com/akeneo/pim-enterprise-dev/pull/9430

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php [hunk 1]
```diff
@@ -342,6 +343,16 @@ protected function putAction(Request $request, $identifier)
...
 
         $data = json_decode($request->getContent(), true);
+
+        try {
+            $this->eventDispatcher->dispatch(
+                JobInstanceEvents::PRE_SAVE,
+                new GenericEvent($jobInstance, ['data' => $data])
+            );
+        } catch (JobInstanceCannotBeUpdatedException $e) {
+            return new JsonResponse(['message' => 'pim_import_export.entity.job_instance.flash.update.fail_empty_permission'], 400);
```
Comment (mmetayer) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php:353]: no, the error message should come from the exception (not really clean, but the permissions are a EE feature, your controller is not aware of them)

Comment (pierrejolly) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php:353]: Indeed, I fix that

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Resources/translations/jsmessages.en_US.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@ pim_import_export:
                 update:
                     success: The job profile has been successfully updated.
                     fail: The job profile could not be updated.
+                    fail_empty_permission: You must select at least one allowed to edit job
```
Comment (mmetayer) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Resources/translations/jsmessages.en_US.yml:26]: now you have to move this translation to EE also :wink:

File: CHANGELOG.md [hunk 3]
```diff
@@ -25,6 +25,7 @@
 - PIM-9360: Fix PHP Warning raised in PriceComparator
 - PIM-9370: Fixes page freezing with a big number of attribute options
 - PIM-9391: Filter empty prices and measurement values
+- PIM-9393: Add error message on job instance when permissions edit is empty
```
Comment (mmetayer) [inline CHANGELOG.md:28]: To remove, the on in EE is enough ;)

**Pull request merged**